simulator/runner/bugbase.rs

```diff
@@ -0,0 +1,241 @@
+use std::{
+    collections::HashMap,
+    io::{self, Write},
+    path::PathBuf,
+    process::Command,
+};
+
+use crate::{InteractionPlan, Paths};
+
+/// A bug is a run that has been identified as buggy.
+#[derive(Clone)]
+pub(crate) enum Bug {
+    Unloaded { seed: u64 },
+    Loaded { seed: u64, plan: InteractionPlan },
+}
+
+impl Bug {
+    /// Check if the bug is loaded.
+    pub(crate) fn is_loaded(&self) -> bool {
+        match self {
+            Bug::Unloaded { .. } => false,
+            Bug::Loaded { .. } => true,
+        }
+    }
+
+    /// Get the seed of the bug.
+    pub(crate) fn seed(&self) -> u64 {
+        match self {
+            Bug::Unloaded { seed } => *seed,
+            Bug::Loaded { seed, .. } => *seed,
+        }
+    }
+}
+
+/// Bug Base is a local database of buggy runs.
+pub(crate) struct BugBase {
+    /// Path to the bug base directory.
+    path: PathBuf,
+    /// The list of buggy runs, uniquely identified by their seed
+    bugs: HashMap<u64, Bug>,
```

Comment:
I would suggest we make this indexable by both `seed` and `hash`. Same seed could not reproduce on a different branch.

Reply:
That was actually the first thing I did :/ but then the logic got too complicated with `--load` and generally the decisions of what to do when run from a different branch.

What about keeping a list of reproducing/non-reproducing hashes within the Bug structure? That way we can also run trials across different branches and have a documentation of the bug locality?

Reply:
I'm okay with both, I don't know what is the best way but I like one where we also store some hash to come back too if we need to.